Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -5,120 +5,142 @@
         // move self and increment resetCount to prevent transform interpolation
         AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTranslation, startingPosition);
         GetNetworkTransformComponentController()->ModifyResetCount()++;
 
-        RigidBodyBus::Event(GetEntityId(), &RigidBodyBus::Events::EnablePhysics);
-        RigidBodyBus::Event(GetEntityId(), &RigidBodyBus::Events::SetLinearVelocity, direction * aznumeric_cast<float>(m_speed));
-
-        GameplayEffectsNotificationBus::Broadcast(&GameplayEffectsNotificationBus::Events::OnPositionalEffect,
-            SoundEffect::EnergyBallTrapProjectile, GetEntity()->GetTransform()->GetWorldTranslation());
+        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::EnablePhysics);
+        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::SetLinearVelocity, direction * GetGatherParams().m_travelSpeed);
     }
-#endif
 
-    void EnergyBallComponentController::HideEnergyBall()
+    void EnergyBallComponentController::CheckForCollisions()
     {
-        Physics::RigidBodyRequestBus::Event(GetEntityId(), &Physics::RigidBodyRequestBus::Events::DisablePhysics);
-
-        // move self and increment resetCount to prevent transform interpolation
-        AZ::TransformBus::Event(GetEntityId(), &AZ::TransformBus::Events::SetWorldTranslation, AZ::Vector3::CreateAxisZ(-1000.f));
-        GetNetworkTransformComponentController()->ModifyResetCount()++;
-    }
+        const AZ::Vector3& position = GetEntity()->GetTransform()->GetWorldTM().GetTranslation();
+        const HitEffect& effect = GetHitEffect();
 
-    void EnergyBallComponentController::OnCollisionBegin(const AzPhysics::CollisionEvent& collisionEvent)
-    {
-        if (collisionEvent.m_body2)
+        IntersectResults results;
+        const ActivateEvent activateEvent{ GetEntity()->GetTransform()->GetWorldTM(), position, Multiplayer::InvalidNetEntityId, Multiplayer::InvalidNetEntityId };
+        GatherEntities(GetGatherParams(), activateEvent, m_filteredNetEntityIds, results);
+        if (!results.empty())
         {
-            AZ::Entity* target = nullptr;
-            AZ::ComponentApplicationBus::BroadcastResult(target,
-                &AZ::ComponentApplicationBus::Events::FindEntity, collisionEvent.m_body2->GetEntityId());
-            if (target)
+            bool shouldTerminate = false;
+            for (const IntersectResult& result : results)
             {
-#if AZ_TRAIT_SERVER
-                if (NetworkHealthComponent* health = target->FindComponent<NetworkHealthComponent>())
+                Multiplayer::ConstNetworkEntityHandle handle = Multiplayer::GetMultiplayer()->GetNetworkEntityManager()->GetEntity(result.m_netEntityId);
+                if (handle.Exists())
                 {
-                    health->SendHealthDelta(aznumeric_cast<float>(-m_armorDamage));
+                    shouldTerminate = true;
```

Comment:
does this mean an energy ball will not terminate if it hits some world geometry or terrain that doesn't have a network entity handle?

Reply:
do we have some kind of range for these to prevent them from living/ticking forever?

Reply:
Yup, that's correct.. it bounces because of the rigid component, which might allow for some interesting gameplay behaviours.